pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -799,4 +806,30 @@ func (ssc *defaultStatefulSetControl) updateStatefulSetStatus(
 	return nil
 }
 
+// wasStatefulSetPreviouslyHealthy determines if a StatefulSet was previously healthy
+// to distinguish between expected unavailability (during initial rollouts, scaling)
+// and unexpected unavailability that should be reported as violations
+func wasStatefulSetPreviouslyHealthy(set *apps.StatefulSet, status *apps.StatefulSetStatus) bool {
```

Comment:
This doesn't seem to be used, didn't you want to combine that with `inconsistentStatus` call in line 745?